src/ui/__tests__/ui_exports_replace_injected_vars.js

```diff
@@ -32,17 +32,24 @@ describe('UiExports', function () {
         server: { port: 0 }, // pick a random open port
         logging: { silent: true }, // no logs
         optimize: { enabled: false },
-        uiSettings: { enabled: false },
         plugins: {
           paths: [resolve(__dirname, './fixtures/test_app')] // inject an app so we can hit /app/{id}
         },
       });
 
       await kbnServer.ready();
-      kbnServer.status.get('ui settings').state = 'green';
-      kbnServer.server.decorate('request', 'getUiSettingsService', () => {
-        return { getDefaults: noop, getUserProvided: noop };
-      });
+
+      // TODO: hopefully we can add better support for something
+      // like this in the new platform
+      kbnServer.server._requestor._decorations.getUiSettingsService = {
```

Comment:
😢 when we upgrade `sinon` to 2.0.0 we can probably do another hack. Isn't really better, just another option:
```js
const decorateStub = sinon.stub(Hapi.Server.super_.prototype, 'decorate');
decorateStub.withArgs('request', 'getUiSettingsService').returns({
	getDefaults: noop,
	getUserProvided: noop
});
decorateStub.callThrough();
```

Reply:
Huh, stub.callThrough() is like spy + optional stubbing? interesting...

Reply:
Yeah, something like this. I played with it yesterday a bit, but didn't manage to make it work in this example with `prototype` - either I did something wrong or it just doesn't set proper context for methods stubbed through prototype.